akka-docs/src/main/paradox/discovery/index.md

```diff
@@ -0,0 +1,209 @@
+# Discovery
+
+@@@ warning
+
+This module is currently marked as @ref:[may change](../common/may-change.md)
+This means that API or semantics can
+change without warning or deprecation period and it is not recommended to use
+this module in production.
+
+@@@
+
+Akka Discovery provides an interface around various ways of locating services, such as DNS
+or using configuration or key-value stores like zookeeper, consul. The built in methods are:
+
+* Configuration
+* DNS
+* Aggregate
+
+In addition `akka-management` contains methods for:
+
+* Kubernetes API
+* AWS
+* Consul
+* Marathon API
+
+Discovery used to be part of `akka-management` but has become an Akka module as of 2.5.19 of Akka and version 0.20
+of `akka-management`.
+
+## Dependency
+
+@@dependency[sbt,Gradle,Maven] {
+  group="com.typesafe.akka"
+  artifact="akka-discovery_2.12"
+  version="$akka.version$"
+}
+
+## What is Service Discovery
+
+Akka's Discovery talks specifically about discovering hosts and ports that relate to some
+logical name.
+
+Discovery is done in a method agnostic way:
+
+Scala
+:  @@snip [CompileOnlySpec.scala](/akka-discovery/src/test/scala/doc/akka/discovery/CompileOnlySpec.scala) { #loading }
+
+Java
+:  @@snip [CompileOnlyTest.java](/akka-discovery/src/test/java/jdoc/akka/discovery/CompileOnlyTest.java) { #loading }
+
+A `Lookup` contains a mandatory `serviceName` and an optional `portName` and `protocol`. How these are interpreted is discovery 
+method dependent e.g.DNS does an A/AAAA record query if any of the fields are missing and an SRV query for a full look up:
+
+Scala
+:  @@snip [CompileOnlySpec.scala](/akka-discovery/src/test/scala/doc/akka/discovery/CompileOnlySpec.scala) { #simple }
+
+Java
+:  @@snip [CompileOnlyTest.java](/akka-discovery/src/test/java/jdoc/akka/discovery/CompileOnlyTest.java) { #simple }
+
+
+`portName` and `protocol` are optional and their meaning is interpreted by the method.
+
+Scala
+:  @@snip [CompileOnlySpec.scala](/akka-discovery/src/test/scala/doc/akka/discovery/CompileOnlySpec.scala) { #full }
+
+Java
+:  @@snip [CompileOnlyTest.java](/akka-discovery/src/test/java/jdoc/akka/discovery/CompileOnlyTest.java) { #full }
+
+Port can be used when a service opens multiple ports e.g. a HTTP port and an Akka remoting port.
+
+## Discovery Method: DNS
+
+DNS discovery maps `Lookup` queries as follows:
+
+* `serviceName`, `portName` and `protocol` set: SRV query in the form: `_port._protocol._name` Where the `_`s are added.
+* Any query  missing any of the fields is mapped to a A/AAAA query for the `serviceName`
+
+The mapping between Akka service discovery terminology and SRV terminology:
+
+* SRV service = port
+* SRV name = serviceName
+* SRV protocol = protocol
+
+Configure it to be used as discovery implementation in your `application.conf` and `async-dns` to be uses
+as the Akka DNS resolver:
```

Comment:
changing to async-dns had other possibly undesired consequences and now it loads async dns anyway, so should it be removed from this docs, or come with warnings?

Reply:
Removed the note to make it the akka resolver